### PR TITLE
javac: add -d and multiple files examples

### PR DESCRIPTION
**Pull request by mshmoustafa**

### CODE REVIEW

File: pages/common/javac.md [hunk 1]
```diff
@@ -5,3 +5,19 @@
 - Compile a .java file:
 
 `javac {{filename.java}}`
+
+- Compile two or more .java files:
```
Comment (waldyrious) [inline pages/common/javac.md]: Suggestion: "two or more" --> "several"

File: pages/common/javac.md [hunk 2]
```diff
@@ -5,3 +5,19 @@
...
+
+`javac {{filename1.java}} {{filename2.java}} {{filename3.java}}`
+
+- Compile all .java files in current directory:
+
+`javac {{*.java}}`
+
+- Compile a .java file and place class file in a specific directory:
```
Comment (waldyrious) [inline pages/common/javac.md]: Suggestion: "...and place the resulting class file..."

File: pages/common/javac.md [hunk 3]
```diff
@@ -5,3 +5,19 @@
...
+
+`javac -d {{path/to/some/directory}} {{filename.java}}`
+
+- Compile two or more .java files and place class files in a specific directory:
```
Comment (waldyrious) [inline pages/common/javac.md]: This one is unnecessary IMO, but if you prefer to keep it, use "several" here too.

Comment (mshmoustafa) [inline pages/common/javac.md]: I agree with your suggestions, I'll make the changes tonight when I'm in front of a computer. Do you think a command line beginner would be able to combine the previous examples to come up with the last one?

-- Muhammad-Sharif Moustafa

Comment (waldyrious) [inline pages/common/javac.md]: Yeah, I don't think that's asking too much. At best we could provide an example combining two options (where at least one of them is new), but at some point this ceases being a primer to the specific command and becomes a primer to the command line, so we have to draw the line somewhere. Feel free to make a judgment call :)

**Comment by mshmoustafa** [comment]
@waldyrious Hey, school started again so I haven't been able to push my edits upstream.  However, my fork is now behind upstream by some 200 commits.  I did  ```git checkout master``` and 
```git pull upstream master```, but should I rebase my topic branch (javac-examples) on top of my master branch?  Also, I still have makefiles in my fork, but you no longer use them, which is another reason I'd like to figure this out.  Sorry, I'm still new to Github, so I'm still trying to learn the protocols you are using.

**Comment by rubenvereecken** [comment]
The rebase should forward you all the way to where we are now and even remove the makefiles. _should_ :). At that point you can push again so your remote, the stuff we're seeing, is up to date as well.

**Comment by mshmoustafa** [comment]
Darn, I wish I saw your comment earlier.  I did a ```git push --force``` to squash my commits, and now I have 7 commits that are obviously not mine... I'm assuming it doesn't look good anymore :/

**Comment by mshmoustafa** [comment]
Although my commit of javac.md looks like it's supposed to, if you ignore all the other 6 commits that showed up.

**Comment by mshmoustafa** [comment]
To be clear, here is the complete sequence of steps I did:
```git checkout master```
```git pull upstream master```
```git push origin master```
```git checkout javac-examples```
```git rebase master```
```git push origin javac-examples```
At this point I found a lot of merge commits, so I did:
```git rebase -i HEAD~7```
I deleted the merge commits and some of my other similar javac commits, but I was unaware that I 'picked' some other people's commits.  Then I did ```git push origin javac-examples --force```.  Is my javac-examples branch borked, or can I fix it?

**Comment by rubenvereecken** [comment]
It's been a while since I've wielded my Git staff this way but you should be able to cherry-pick your way out of trouble.

1. Get rid of all those commits or just make sure you have a clean branch that is the same as master. `git reset --hard HEAD~7` to remove last 7 commits.
2. Cherry pick the wanted commits: `git cherry-pick 8511e5f`, where that sha is just the one belonging to the commit you want. It should be that one.

You should now have that same branch + the one commit you were looking for.

**Comment by mshmoustafa** [comment]
Yay, we fixed it! Thanks! If my commit looks good, I'd appreciate it if you pull it in. :) It's my first pull request you know, very exciting! I learned a lot about Git too.

**Comment by ostera** [comment]
Merging since @rubenvereecken already gave greenlight if it was rebased/cherry-picked (and it was!). Thanks @mshmoustafa!

**Comment by waldyrious** [comment]
Great work @mshmoustafa!

**Comment by rubenvereecken** [comment]
That's some hardcode Gittery for a first PR, nice one @mshmoustafa.

**Pull request merged**